pyasic/miners/backends/auradine.py

```diff
@@ -193,6 +195,49 @@ async def send_config(self, config: MinerConfig, user_suffix: str = None) -> Non
         for key in conf.keys():
             await self.web.send_command(command=key, **conf[key])
 
+    async def upgrade_firmware(self, *, url: str = None, file_path: str = None, version: str = "latest") -> bool:
+        """
+        Upgrade the firmware of the Auradine device.
+
+        Args:
+            url (str): The URL to download the firmware from.
+            file_path (str): The local file path of the firmware to be uploaded.
+            version (str): The version of the firmware to upgrade to.
+
+        Returns:
+            bool: True if the firmware upgrade was successful, False otherwise.
+        """
+        try:
```

Comment:
This whole function should use the `self.web` attribute.

Reply:
Done.